### ISSUE TITLE
i_mdi_dictionary escape sequence corrupts rest of the file

### ISSUE THREAD

**Issue by onlinespending** [ryanoasis/nerd-fonts#745]
#### 🗹 Requirements

- [x] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my issue and found nothing related and/or helpful
- [x] I have searched the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting) for help
- [x] I have searched the [Wiki](https://github.com/ryanoasis/vim-devicons/wiki) for help

#### 🎯 Subject of the issue

the Material Design dictionary icon uses an escaped backslash which seems to mess up the renderings of all subsequent characters.

This is the mdi_dictionary character using 'cat -A'.
i='M-oM-,M-^\\'

The \\' likely causing issues. This issue can be seen simply using the test-fonts.sh script. Or in the lib/i_material.sh script. Is there a way to workaround this issue?

Thank you.

#### 🔧 Your Setup

- _Which font are you using (e.g. `Anonymice Powerline Nerd Font Complete.ttf`)?_Hack Nerd Font Complete.ttf
- _Which terminal emulator are you using (e.g. `iterm2`, `urxvt`, `gnome`, `konsole`)?_gnome and others
- _Are you using OS X, Linux or Windows? And which specific version or distribution?_Linux CentOS 8

**Comment by Finii** [comment]
I can not really see what @onlinespending sees, but there seems to be something wrong...

From what I see with vim there is a `'` missing and not escaped?

![image](https://user-images.githubusercontent.com/16012374/147781333-795c87f8-000d-4ff5-b1b7-c5d8afc7be5b.png)

Whatever, I have no idea if this is autogenerated or what.

**Comment by Finii** [comment]
But looking at the hexdump it seems that is a vim rendering issue.

The `i_mdi_format_horizontal_align_left` value looks ligit.

![image](https://user-images.githubusercontent.com/16012374/147782126-cd03d4ee-6b02-48f0-bce4-38ac0915bfb4.png)

**Comment by onlinespending** [comment]
Yes that's what I see. But I believe the unmatched ' is due to the escaped backslash that comes right before the dictionary icons closing '

Interestingly, if I remove that line (and even a few adjacent lines) and source the i_material.sh from scratch, the variables set below dictionary are blank when echoing at the shell prompt

**Comment by Finii** [comment]
`test-fonts` looks ok...

![image](https://user-images.githubusercontent.com/16012374/147783258-a552a995-8e2b-4f52-be92-b4775a009054.png)

There are some width issues because I use a non-Mono font, but ...

Maybe you should show how it looks wrong.

**Comment by Finii** [comment]
> But I believe the unmatched ' is due to the escaped backslash that comes right before the dictionary icon's closing '

Well, looking on the hexdump, there is no escaped backlash whatever.

![image](https://user-images.githubusercontent.com/16012374/147783450-9bffa632-3f23-4b14-a75a-d75ee203e371.png)

**Comment by onlinespending** [comment]
That’s what $9c is, which won’t display as an ASCII character in your hex viewer. However, I don’t believe this is an issue. I had tried multiple shells and hacked a few things, which I think caused some of the weirdness I was observing. But after restarting my shell and using gnome-terminal, I can source the i_*.sh files and echo even those characters that are defined after the dictionary. Still finding issues with other shells, but that’s not nerd-fonts problem. Closing this. Thank you for your help.
